esda/moran.py

```diff
@@ -506,9 +506,9 @@ def by_col(
         )
 
 
-def Moran_BV_matrix(variables, w, permutations=0, varnames=None):
+class Moran_BV_matrix:
```

Comment:
This is a hard breaking change. `Moran_BV_matrix` currently returns a result, with this it no longer does and needs to be retrieved via `Moran_BV_matrix.matrix`. Why have you opted for changing this to a class?

Reply:
I did because I saw a this issue with enhancement tag in it [](https://github.com/pysal/esda/issues/31), if it does not seem right to you, I will revert it back to function instead of class

Reply:
I did this because I saw the issue with enhancement tag with it [https://github.com/pysal/esda/issues/31](url). If this does not seem right to you, I will revert it back to function.
Thank you

Reply:
We can do that as proposed in #31 but there needs to be a deprecation period, otherwise it may break code in downstream. I suggest leaving it out of this PR and we can tackle it separately.